walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,104 @@
+            }
+        }
+
+        if (differentAttendees == 1) {
+            return Arrays.asList(TimeRange.WHOLE_DAY);
+        }
+        return available;
+    }
+
+    public List<TimeRange> remove(Collection<TimeRange> collectionRanges, TimeRange rangeToRemove) {
+        int index = -1;
+        ArrayList<TimeRange> ranges = new ArrayList<>(collectionRanges);
+        for (TimeRange range : ranges) {
+            
+            index++;
+
+            // Case 1: range is fully contained in rangeToRemove
+            //   |---|            range
+            // |--------|         rangeToRemove
+            if (range.start() >= rangeToRemove.start() && range.end() <= rangeToRemove.end()) {
+                ranges.remove(index);
+                index--;
+                continue;
+            }
+
+            // Case 2: range starts before but end while rangeToRemove is happening
+            //   |------|         range
+            //      |--------|    rangeToRemove
+            if (range.start() < rangeToRemove.start() && range.end() >= rangeToRemove.start() && range.end() <= rangeToRemove.end()) {
+                TimeRange newRange = TimeRange.fromStartEnd(range.start(), rangeToRemove.start(), false);
+                ranges.set(index, newRange);
+                continue;
+            }
+
+            // Case 3: range ends after but starts while rangeToRemove is happening
+            //   |--------|       range
+            // |------|           rangeToRemove
+            if (range.start() >= rangeToRemove.start() && range.start() <= rangeToRemove.end() && range.end() > rangeToRemove.end()) {
+                TimeRange newRange = TimeRange.fromStartEnd(rangeToRemove.end(), range.end(), false);
+                ranges.set(index, newRange);
+                continue;
+            }
+
+            // Case 4: range fully covers rangeToRemove
+            // |---------|       range
+            //   |----|          rangeToRemove
+            if (range.start() < rangeToRemove.start() && range.end() > rangeToRemove.end()) {
+                TimeRange leftRange = TimeRange.fromStartEnd(range.start(), rangeToRemove.start(), false);
+                TimeRange rightRange;
+                if (range.end() >= TimeRange.END_OF_DAY) {
+                   rightRange = TimeRange.fromStartEnd(rangeToRemove.end(), TimeRange.END_OF_DAY, true);
+                } else {
+                    rightRange = TimeRange.fromStartEnd(rangeToRemove.end(), range.end(), true);
+                }
+                ranges.set(index, leftRange);
+                ranges.add(index+1, rightRange);
+                break;
+            }
+
+        }
+
+        return ranges;
+    }
+
+    public boolean attendeesOverlap(Set<String> setA, Set<String> setB) {
+        Set<String> intersection = new HashSet<>(setA);
+        intersection.retainAll(setB);
+        if (intersection.isEmpty()) {
+            return false;
+        } else {
+            return true;
+        }
```

Comment:
This structure is an anti-pattern ;-)

```
if (..something..) {
 return true;
} else {
 return false;
}
```

It should always be written as `return !intersection.isEmpty();`